File: python/lists/linked-list/to-do-list/task.py
class TaskStatus:
    PENDING = "Pending"
    COMPLETED = "Completed"

class Task:
    def __init__(self, id, description, tag, status=TaskStatus.PENDING):
        self.id = id
        self.description = description
        self.tag = tag
        self.status = status

    def get_id(self):
        return self.id

    def get_description(self):
        return self.description

    def set_description(self, description):
        self.description = description

    def get_tag(self):
        return self.tag

    def set_tag(self, tag):
        self.tag = tag

    def get_status(self):
        return self.status

    def set_status(self, status):
        self.status = status

    def __str__(self):
        return f"Task {self.id}: {self.description} | Tag: {self.tag} | Status: {self.status}"

# Example:
# task = Task(id=1, description="Example Task", tag="Work")
# print(task)

File: javascript/lists/linked-list/to-do-list/task.js
export const TaskStatus = {
    PENDING: "Pending",
    COMPLETED: "Completed",
};

export class Task {
    constructor(id, description, tag, status = TaskStatus.PENDING) {
        this.id = id;
        this.description = description;
        this.tag = tag;
        this.status = status;
    }

    getId = () => {
        return this.id;
    };

    getDescription = () => {
        return this.description;
    };

    setDescription = (description) => {
        this.description = description;
    };

    getTag = () => {
        return this.tag;
    };

    setTag = (tag) => {
        this.tag = tag;
    };

    getStatus = () => {
        return this.tag;
    };

    setStatus = (status) => {
        this.status = status;
    };
    
    toString() {
        return `Task ${this.id}: ${this.description} | Tag: ${this.tag} | Status: ${this.status}`;
    };
}